frappe/core/doctype/user/user.py

```diff
@@ -325,17 +325,27 @@ def on_trash(self):
 		frappe.db.sql("""delete from `tabDocShare` where user=%s""", self.name)
 
 		# delete messages
-		frappe.db.sql("""delete from `tabCommunication`
-			where communication_type in ('Chat', 'Notification')
-			and reference_doctype='User'
-			and (reference_name=%s or owner=%s)""", (self.name, self.name))
+		parents = frappe.db.sql("""select parent from `tabDynamic Link`
+				where `tabDynamic Link`.link_doctype='User'
+				and (`tabDynamic Link`.link_name=%s or `tabDynamic Link`.link_owner=%s)
```

Comment:
Communications with `owner` as `user` and Links with `reference_name` as `user` should be fetched here. This only checks for `link_owner` and not `owner`.